### ISSUE TITLE
Android compilation broken after sRGB additions

### ISSUE THREAD

**Issue by gordonmcshane** [SFML/SFML#1079]
Workaround:
https://github.com/gordonmcshane/SFML/commit/c89d1a987bc8358ea87c56c5aac5f25e4fec8e2d

**Comment by eXpl0it3r** [comment]
Would been nice if you add some more information or even better follow the [Contribution Guidelines](http://www.sfml-dev.org/contribute.php)...

**Comment by gordonmcshane** [comment]
Apologies. I don't spend a lot of time in SFML code base and didn't want to be presumptuous in thinking my solution was sufficiently correct enough not to break other aspects of SFML which is why there was  no pull request. My intent was just to unblock anyone while a better fix was worked out. I should have left a log though:

`[ 79%] Building CXX object src/SFML/Graphics/CMakeFiles/sfml-graphics.dir/RenderTextureImplDefault.cpp.o
In file included from /Users/gordon/Projects/SFML/src/SFML/Graphics/GLCheck.hpp:32:0,
                 from /Users/gordon/Projects/SFML/src/SFML/Graphics/Texture.cpp:30:
/Users/gordon/Projects/SFML/src/SFML/Graphics/Texture.cpp: In member function 'bool sf::Texture::create(unsigned int, unsigned int)':
/Users/gordon/Projects/SFML/src/SFML/Graphics/GLExtensions.hpp:118:55: error: 'GL_EXT_sRGB' was not declared in this scope
     #define GLEXT_texture_sRGB                        GL_EXT_sRGB
                                                       ^
/Users/gordon/Projects/SFML/src/SFML/Graphics/Texture.cpp:190:31: note: in expansion of macro 'GLEXT_texture_sRGB'
     static bool textureSrgb = GLEXT_texture_sRGB;
                               ^
In file included from /Users/gordon/Projects/SFML/src/SFML/Graphics/Texture.cpp:30:0:
/Users/gordon/Projects/SFML/src/SFML/Graphics/GLExtensions.hpp:119:55: error: 'GL_SRGB8_ALPHA8_EXT' was not declared in this scope
     #define GLEXT_GL_SRGB8_ALPHA8                     GL_SRGB8_ALPHA8_EXT
                                                       ^
/Users/gordon/Projects/SFML/src/SFML/Graphics/GLCheck.hpp:51:28: note: in definition of macro 'glCheck'
     #define glCheck(expr) (expr)
                            ^
/Users/gordon/Projects/SFML/src/SFML/Graphics/Texture.cpp:213:54: note: in expansion of macro 'GLEXT_GL_SRGB8_ALPHA8'
     glCheck(glTexImage2D(GL_TEXTURE_2D, 0, (m_sRgb ? GLEXT_GL_SRGB8_ALPHA8 : GL_RGBA), m_actualSize.x, m_actualSize.y, 0, GL_RGBA, GL_UNSIGNED_BYTE, NULL));
                                                      ^
make[2]: *** [src/SFML/Graphics/CMakeFiles/sfml-graphics.dir/Texture.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [src/SFML/Graphics/CMakeFiles/sfml-graphics.dir/all] Error 2
make: *** [all] Error 2`

If you think my changes are at least a decent starting point, let me know and I will submit a pull request per the contribution guidelines to be reviewed.

**Comment by eXpl0it3r** [comment]
That's why encourage people in the Contribution Guidelines to first use the forum and reserve the issue tracker for confirmed issues.

What Android NDK are you using? What OS are you on? etc.

**Comment by gordonmcshane** [comment]
Host: OSX 10.11.4
Target: armeabi-v7a (no thumb)

NDK r11b (which removed 4.8 and clang 3.5)
CMake 3.5.1

Toolchain: prebuilt gcc 4.9
STL: gnustl_shared

**Comment by mantognini** [comment]
@gordonmcshane Could you provide a PR for the 1st patch? (And only that one so that we can discuss, if needed, the other one separately. For the second one, you should probably explain exactly what's your setup and the issue you're facing on the forum.)

**Comment by MarioLiebisch** [comment]
Finally had some time to check this. The linked patch is not needed.

The correct way to do this is forcing a minimum NDK API level of 12 (Android 3.1.x or later). What do you think?

If we're using the provided patch, we'll lose that support even for versions that support it.

**Comment by mantognini** [comment]
> The correct way to do this is forcing a minimum NDK API level of 12 (Android 3.1.x or later). What do you think?

#1085 seems to do that but with min = 21.

**Comment by MarioLiebisch** [comment]
Which is overkill. API 21 is 5.0 and newer only.

**Comment by binary1248** [comment]
```diff
diff --git a/src/SFML/Graphics/GLExtensions.hpp b/src/SFML/Graphics/GLExtensions.hpp
index c3333b7..c9ed451 100644
--- a/src/SFML/Graphics/GLExtensions.hpp
+++ b/src/SFML/Graphics/GLExtensions.hpp
@@ -34,6 +34,14 @@

     #include <SFML/OpenGL.hpp>

+    // Since SFML enforces Android API level 9 or greater,
+    // <GLES2/gl2ext.h> will always be available on Android
+    #ifdef SFML_SYSTEM_ANDROID
+
+        #include <GLES2/gl2ext.h>
+
+    #endif
+
     // SFML requires at a bare minimum OpenGL ES 1.0 capability
     // Some extensions only incorporated by 2.0 are also required
     // OpenGL ES 1.0 is defined relative to OpenGL 1.3
@@ -112,8 +120,13 @@
     #define GLEXT_GL_INVALID_FRAMEBUFFER_OPERATION    GL_INVALID_FRAMEBUFFER_OPERATION_OES

     // Core since 3.0 - EXT_sRGB
-    #define GLEXT_texture_sRGB                        GL_EXT_sRGB
-    #define GLEXT_GL_SRGB8_ALPHA8                     GL_SRGB8_ALPHA8_EXT
+    #ifdef GL_EXT_sRGB
+        #define GLEXT_texture_sRGB                        GL_EXT_sRGB
+        #define GLEXT_GL_SRGB8_ALPHA8                     GL_SRGB8_ALPHA8_EXT
+    #else
+        #define GLEXT_texture_sRGB                        false
+        #define GLEXT_GL_SRGB8_ALPHA8                     0
+    #endif

 #else


```

**Comment by eXpl0it3r** [comment]
Is there a way to check the NDK API level and return a verbose message if the minimal version isn't met?

**Comment by binary1248** [comment]
CMake already does this. If the user specifies some value smaller than 9, configuration will fail.

**Comment by MarioLiebisch** [comment]
Do we have any feature that requires the sRGB extension? If not, binary's solution would work (without including the v2 header).

**Comment by eXpl0it3r** [comment]
What do you mean by feature? Does the normal sRGB support not require the extension?

**Comment by binary1248** [comment]
There is no typical "feature" that requires sRGB support. sRGB support is the feature itself, when it is available. If the NDK against which the library is built doesn't expose it then it just won't be available for the user to use (i.e. calling the functions won't do anything).

**Comment by eXpl0it3r** [comment]
Bump

**Comment by h0tw1r3** [comment]
Here's what I am using.

```diff
diff --git a/include/SFML/OpenGL.hpp b/include/SFML/OpenGL.hpp
index a5d964f..a62ea81 100644
--- a/include/SFML/OpenGL.hpp
+++ b/include/SFML/OpenGL.hpp
@@ -67,7 +67,12 @@
 
     #include <GLES/gl.h>
     #include <GLES/glext.h>
+    #include <GLES2/gl2ext.h>
 
+    #if __ANDROID_API__ < 21
+        #define GL_EXT_sRGB false
+        #define GL_SRGB8_ALPHA8_EXT 0
+    #endif
 #endif
 
 
```
